rules/DeadCode/Rector/Assign/RemoveUnusedVariableAssignRector.php

```diff
@@ -152,6 +169,31 @@ private function isUsed(Assign $assign, Variable $variable): bool
         return false;
     }
 
+    private function isInCompact(FuncCall $funcCall, Variable $variable): bool
+    {
+        if (! $this->isName($funcCall, 'compact')) {
+            return false;
+        }
+
+        $variableName = $variable->name;
+        if (! is_string($variableName)) {
+            return false;
+        }
+
+        $args = $funcCall->args;
+        foreach ($args as $arg) {
+            if (! $arg->value instanceof String_) {
+                continue;
+            }
+
+            if ($arg->value->value === $variableName) {
+                return true;
+            }
+        }
+
+        return false;
+    }
```

Comment:
This could be useful in few more places. Could you extra it to own service, e.g `CompactFuncCallAnalyzer`?

Reply:
implemented